IPython/core/completer.py

```diff
@@ -608,20 +609,15 @@ def __init__(self, shell=None, namespace=None, global_namespace=None,
           both Python scopes are visible.
 
         use_readline : bool, optional
-          If true, use the readline library.  This completer can still function
-          without readline, though in that case callers must provide some extra
-          information on each call about the current line."""
+          DEPRECATED, ignored.
+        """
 
         self.magic_escape = ESC_MAGIC
         self.splitter = CompletionSplitter()
 
-        # Readline configuration, only used by the rlcompleter method.
         if use_readline:
-            # We store the right version of readline so that later code
-            import IPython.utils.rlineimpl as readline
-            self.readline = readline
-        else:
-            self.readline = None
+            warnings.warn('The use_readline parameter is deprecated since IPython 6.0.',
```

Comment:
Perhaps nitpicky:

``'The use_readline parameter is deprecated and ignored since IPython 6.0.'``